kasa/discover.py

```diff
@@ -236,3 +236,24 @@ def _get_device_class(info: dict) -> Type[SmartDevice]:
             return SmartBulb
 
         raise SmartDeviceException("Unknown device type: %s" % type_)
+
+    @staticmethod
+    def _create_device_from_discovery_info(ip, info: dict) -> SmartDevice:
+        """Create a SmartDevice for device described by passed data."""
+        device_class = Discover._get_device_class(info)
+        device = device_class(ip)
+        device.update_from_discovery_info(info)
+        return device
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO)
+    loop = asyncio.get_event_loop()
+
+    async def _on_device(dev):
+        await dev.update()
+        _LOGGER.info("Got device: %s", dev)
+
+    devices = loop.run_until_complete(Discover.discover(on_discovered=_on_device))
+    for ip, dev in devices.items():
+        print(f"[{ip}] {dev}")
```

Comment:
```suggestion
```
Not needed here. https://python-kasa.readthedocs.io/en/latest/discover.html#kasa.Discover has an example on how to use `on_discovered`, do you think that's enough or should the repository contain a separate examples directory hosting examples?